app/controllers/users/sessions_controller.rb

```diff
@@ -29,10 +29,15 @@ def new
     def create
       track_authentication_attempt(auth_params[:email])
 
+      clear_session_bad_password_count_if_window_expired
+      return process_locked_out_session if session_bad_password_count_max_exceeded?
       return process_locked_out_user if current_user && user_locked_out?(current_user)
 
+      @throttle_password_failure = true
       self.resource = warden.authenticate!(auth_options)
       handle_valid_authentication
+    ensure
+      increment_session_bad_password_count if @throttle_password_failure && !current_user
```

Comment:
I think this can be a local variable instead of an instance variable, since it's only used here and not in another method or template

```suggestion
      throttle_password_failure = true
      self.resource = warden.authenticate!(auth_options)
      handle_valid_authentication
    ensure
      increment_session_bad_password_count if throttle_password_failure && !current_user
```